src/index.ts

```diff
@@ -251,20 +277,31 @@ const PersistedQueriesPlugin: PostGraphilePlugin = {
   },
 
   // For regular HTTP requests
-  "postgraphile:httpParamsList"(paramsList, { options }) {
+  "postgraphile:httpParamsList"(paramsList, { options, req }) {
     return paramsList.map((params: any) => {
       // ALWAYS OVERWRITE, even if invalid; the error will be thrown elsewhere.
-      params.query = persistedOperationFromPayload(params, options) as string;
+      params.query = persistedOperationFromPayload(
+        params,
+        options,
+        options.allowUnpersistedOperations
+          ? options.allowUnpersistedOperations(req, params)
+          : false
```

Comment:
It might make sense to allow `allowUnpersistedOperations: true` as the setting; so this check would become:

```suggestion
        typeof options.allowUnpersistedOperations === "function"
          ? options.allowUnpersistedOperations(req, params)
          : !!options.allowUnpersistedOperations
```